### PR TITLE
Users/bgavril/m95fix

### PR DESCRIPTION
**Pull request by bgavrilMS**
Porting the fix for https://jira.sonarsource.com/browse/TBSTFNP-44 to the M95 branch so that it is included in TFS on-prem Update 3;

There are 3 commits: 

- the first 2 constitute the actual fix and have been cherry-picked from master 
- the 3rd is bumping the sonarqube post test task version (the pre build task version is already bumped)

Note: this also includes some refactoring on how the SonarQubeHelper file is deployed. If you feel it brings too much churn I can create a separate commit that focuses only on the fix.

### CODE REVIEW

File: Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1 [hunk 1]
```diff
@@ -154,14 +154,17 @@ function InvokeGetRestMethod
...
     if ($useAuth)
-    {
-      
+    {      
        $authHeader = CreateBasicAuthHeaderFromEndpoint
-       $allheaders = @{Authorization = $authHeader}   
-    }  
 
-    $response = Invoke-RestMethod $request -Method Get -Headers $allheaders
+       if (![String]::IsNullOrWhiteSpace($authHeader))
```
Comment (bgavrilMS) [inline Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1]: The actual fix is here - the authHeader can be empty (i.e. anon users). In this case, Powershell's Invoke-RestMethod throws a null ref exception;

Workaround for users: don't use anon connections to SonarQube

File: Tests/L0/Common-SonarQubeHelpers/_suite.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+describe('Common-SonarQubeHelpers Suite', function () {
+    this.timeout(20000);
+
+    before((done) => {
+        // init here
+        done();
+    });
+
+    after(function () {
+    });
```
Comment (HSAR) [inline Tests/L0/Common-SonarQubeHelpers/_suite.ts]: Empty function?

Comment (bgavrilMS) [inline Tests/L0/Common-SonarQubeHelpers/_suite.ts]: Yeah, it's boilerplate code, I don't want to touch it.

**Comment by HSAR** [comment]
LGTM

**Pull request merged**